### PR TITLE
fix unnessasry cpu memory usage when training

### PR DESCRIPTION
**Pull request by xiye17**
Use scalar when recording the aggregated training loss throughout the training procedure in `Trainer`.

The previous version uses `torch.Tensor` to store `tr_loss`. Here, tr_loss is accumulating history across your training loop and takes up lots of CPU memory during training. There is an official pytorch post on **NOT to accumulate history across your training loop**. ([https://pytorch.org/docs/stable/notes/faq.html#my-model-reports-cuda-runtime-error-2-out-of-memory](https://pytorch.org/docs/stable/notes/faq.html#my-model-reports-cuda-runtime-error-2-out-of-memory))

 When I tried to train a sentence classififier with batch size 10 on HotpotQA dataset (for document filtering), the program consumed around 120G CPU memory when I trained for 240,000 steps (roughly 3 epoch over all 900,000 HotpotQA documents), which is unnessasry. 

So I think it's better to just use a scalar (as suggested by the official post). Also, the tensor format returned value has never been used in the whole repo. It's a safe fix.

### CODE REVIEW

**Review by LysandreJik — COMMENTED** [review]
We cannot do that as this would harm TPU training. See discussion [here](https://github.com/huggingface/transformers/pull/6999).
